syntax = "proto3";
package courses;
import "locales_types.proto";
import "google/protobuf/timestamp.proto";
import "google/protobuf/field_mask.proto";

message CourseTopic {
  int64 id = 1;
  string name = 2;
  string description = 3;
  CourseTopicShort parent = 4;
  locales.Locale locale = 5;
  bool is_active = 6;
  int32 sort = 7;
  google.protobuf.Timestamp created_at = 8;
  google.protobuf.Timestamp updated_at = 9;
  google.protobuf.Timestamp deleted_at = 10;
}

message CourseTopicShort {
  int64 id = 1;
  string name = 2;
}

message GetCourseTopicRequest {
  int64 course_topic_id = 1;
}

message CreateCourseTopicRequest {
  string name = 1;
  string description = 2;
  int64 parent_id = 3;
  int32 locale_id = 4;
}

message ListCourseTopicsRequest {
  int32 limit = 1;
  int64 offset = 2;
}

message UpdateCourseTopic {
  string name = 2;
  string description = 3;
  int64 parent_id = 4;
  int32 locale_id = 5;
  bool is_active = 6;
  int32 sort = 7;
}

message UpdateCourseTopicRequest {
  int64 course_topic_id = 1;
  UpdateCourseTopic course_topic = 2;
  google.protobuf.FieldMask mask = 3;
}

message ListCourseTopicsResponse {
  repeated CourseTopic course_topics = 1;
}